README.md

```diff
@@ -0,0 +1,400 @@
+
+```c++
+v = -5 * x + y / --z;
+
+x += y++;
+```
+
+Logical operators `&&` and `||` should also have spaces around them.
+This does not apply to unary logical operator `!` which should be written adjacent to the variable it affects.
+
+Similarly, bitwise operators should have spaces around them, except the unary complement (bit inversion) operator `~` which should also be written adjacent to the variable it affects.
+
+Comparison operators ought to have spaces around them too.
+
+```c++
+if (param1 == 0 && param1 > 3 || !param2)
+{
+}
+```
+
+There should be no space around member operators `.` and `->`.
+
+Address-of `&` and de-reference `*` unary operators should be always adjacent to the variable they affect.
+
+When overloading an operator, there should be a space between the `operator` keyword and the operator being overloaded.
+
+```c++
+Subscriber& Subscriber::operator <<(
+        const qos::SubscriberQos& qos)
+{
+    this->qos(qos);
+    return *this;
+}
+```
+
+#### Templates
+
+There should be a space between closing angles `>`.
+
+```c++
+std::map<std::string, std::vector<DataWriterImpl*> > writers_;
+```
+
+#### Line-end Comments
+
+Lines that are non-obvious should get a comment at the end of the line.
+These end-of-line comments should be separated from the code by a space.
+
+### Function Declarations and Definitions
+
+To improve parameter readability, each parameter must be in its own line, even the first one and even if there is only one parameter.
+The line-wrapping indentation is to be used in this case.
+Never indent to the opening parenthesis level, the first parameter, etc.
+
+This does not apply when calling the function.
+In this case the parameters should be in the same line taking into account the space after each comma that separates the parameters.
+
+The closing parenthesis `)` should be adjacent to the last parameter on the list.
+
+The opening and close braces should be each alone in their own line.
+
+Method definition should not have any indentation.
+
+```c++
+void EDPStatic::assignRemoteEndpoints(
+        const ParticipantProxyData& pdata)
+{
+    // Method body
+}
+                                        // Blank line between methods
+// Incorrect
+void EDPStatic::assignRemoteEndpoints(const ParticipantProxyData& pdata)
+{
+    // Method body
+}
+
+bool EDPStatic::newRemoteReader(
+        const GUID_t& participant_guid,
+        const string_255& participant_name,
+        uint16_t user_id,
+        EntityId_t ent_id)
+{
+    // Method body
+}
+```
+
+### Loops and Conditionals
+
+There must always be braces `{ }`, even if there is no need (one line statements or even empty statements).
+Additionally, both the opening and closing braces should be each alone in their own new line.
+
+```c++
+// Correct
+if (!error)
+{
+    return success;
+}
+
+// Incorrect
+if (!error)
+    return success;
+
+// Incorrect
+if (!error) return success;
+```
+
+### Pointer and Reference Expressions
```

Comment:
There is an exception when defining several pointers or references in the same line. In this case the `*` or `&` must be adjacent to the name:
```
int *value, *key;
```

Reply:
Introduced

Reply:
Introduced

Reply:
I disagree with @IkerLuengo as the `exception` breaks the rule, no exceptions should be allowed. Indeed, there should be forbidden to declare several variables in the same line. Always.